contracts/interchain-token-service/src/abi.rs

```diff
@@ -122,49 +126,58 @@ impl Message {
     }
 
     pub fn abi_decode(payload: &[u8]) -> Result<Self, Report<Error>> {
-        ensure!(payload.len() >= 32, Error::MessageDecodeFailed);
+        ensure!(payload.len() >= 32, Error::InsufficientMessageLength);
 
         let message_type = MessageType::abi_decode(&payload[0..32], true)
+            .map_err(Error::AbiDecodeFailed)
```

Comment:
why add additional map?

Reply:
hmm, fair enough, i was wrapping the decoding error for consistency, but I suppose there's no point when changing context to the current Error. Removed